client/state/selectors/eligible-for-domain-to-paid-plan-upsell.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Internal dependencies
+ */
+import {
+	canCurrentUser,
+	isMappedDomainSite,
+	isSiteOnFreePlan,
+} from 'state/selectors/';
+
+/**
+ * Returns true if the current user is eligible for a domain to paid plan upsell for the site
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {?Boolean} True if the user can participate in the domain to paid plan upsell
+ */
+const eligibleForDomainToPaidPlanUpsell = ( state, siteId ) => {
```

Comment:
If we return a boolean, I'd recommend prefixing this with `is` so the reader has a hint. eg `isEligibleForDomainToPaidPlanUpsell`